crates/biome_fs/src/path.rs

```diff
@@ -250,9 +250,7 @@ impl BiomePath {
 
     /// Returns `true` if the path is inside `node_modules`
     pub fn is_dependency(&self) -> bool {
-        self.path
-            .components()
-            .any(|component| component.as_str() == "node_modules")
+        self.path.as_str().contains("node_modules")
```

Comment:
It's not very likely I suppose, but this will ignore directories named `inode_modules`, `node_modules2`, and others too.